### PR TITLE
Migrates aleo-setup to snarkVM

### PR DESCRIPTION
**Pull request by howardwu**
Migrates aleo-setup to snarkVM

### CODE REVIEW

File: setup-utils/src/io/read.rs [hunk 1]
```diff
@@ -62,15 +62,7 @@ impl<R: Read> Deserializer for R {
...
-            UseCompression::No => {
-                if check_for_correctness == CheckForCorrectness::OnlyNonZero
-                    || check_for_correctness == CheckForCorrectness::No
-                {
-                    G::deserialize_unchecked(self)?
-                } else {
-                    G::deserialize_uncompressed(self)?
-                }
-            }
+            UseCompression::No => G::deserialize_uncompressed(self)?,
```
Comment (kellpossible) [inline setup-utils/src/io/read.rs:65]: unchecked variant is no longer required, or it's handled by the new uncompressed function?

File: setup1-verifier/src/utils/authentication.rs [hunk 2]
```diff
@@ -68,7 +64,7 @@ impl AleoAuthentication {
     /// returns a `VerifierError`.
     ///
     pub fn sign(view_key: &ViewKey, message: String) -> Result<String, VerifierError> {
-        let rng = &mut thread_rng();
+        let rng = &mut OsRng;
```
Comment (kellpossible) [inline setup1-verifier/src/utils/authentication.rs:67]: does this have some performance implications?

**Review by kellpossible — COMMENTED** [review]
I don't understand all the technical details of these changes but on the surface it looks good to me

File: setup1-cli-tools/Cargo.toml [hunk 3]
```diff
@@ -16,15 +16,11 @@ path = "src/view_key.rs"
...
 serde_json = "1.0.64"
 structopt = "0.3.21"
 unic-langid = "0.9.0"
 
-[dependencies.snarkos-toolkit]
-git = "https://github.com/AleoHQ/snarkOS"
-rev = "6357695"
-package = "snarkos-toolkit"
-default-features = false
+snarkvm-dpc = { git = "https://github.com/emmorais/snarkVM", rev = "6a409e3" }
```
Comment (ibaryshnikov) [inline setup1-cli-tools/Cargo.toml:26]: git link instead of a version from crates.io

**Review by apruden2008 — APPROVED** [review]
Pending conflict resolution

**Review by ibaryshnikov — APPROVED** [review]
LGTM as soon as the conflicts are resolved

**Comment by howardwu** [comment]
After reviewing the commits since my last one in this branch, some of the dependency usages and changes to rng look non-standard to me. Out of an abundance of precaution, I will be doing the port from my last commit accounting for any changes that we want to keep since then in this branch.

**Comment by howardwu** [comment]
Closing this in favor of PR #317

**Pull request closed without merging**